time/time_test.go

```diff
@@ -8,20 +8,67 @@ import (
 )
 
 func TestDateTime(t *testing.T) {
-	refTime := time.Date(1985, 4, 12, 23, 20, 50, int(520*time.Millisecond), time.UTC)
-
-	dateTime := FormatDateTime(refTime)
-	if e, a := "1985-04-12T23:20:50.52Z", dateTime; e != a {
-		t.Errorf("expected %v, got %v", e, a)
+	cases := map[string]struct {
+		TimeString      string
+		TimeValue       time.Time
+		SymmetricString bool
+	}{
+		"no offset": {
+			TimeString: "1985-04-12T23:20:50.52Z",
+			TimeValue: time.Date(1985, 4, 12, 23, 20, 50, int(520*time.Millisecond),
+				time.UTC),
+			SymmetricString: true,
+		},
+		"with negative offset": {
+			TimeString: "1985-04-12T23:20:50.52-07:00",
+			TimeValue: time.Date(1985, 4, 12, 23, 20, 50, int(520*time.Millisecond),
+				time.FixedZone("-0700", -7*60*60)),
+			SymmetricString: false,
+		},
+		"with positive offset": {
+			TimeString: "1985-04-12T23:20:50.52+07:00",
+			TimeValue: time.Date(1985, 4, 12, 23, 20, 50, int(520*time.Millisecond),
+				time.FixedZone("-0700", +7*60*60)),
+			SymmetricString: false,
+		},
+		"UTC serialize": {
+			TimeString: "1985-04-13T06:20:50.52Z",
+			TimeValue: time.Date(1985, 4, 12, 23, 20, 50, int(520*time.Millisecond),
+				time.FixedZone("-0700", -7*60*60)),
+			SymmetricString: true,
+		},
 	}
 
-	parseTime, err := ParseDateTime(dateTime)
-	if err != nil {
-		t.Fatalf("expected no error, got %v", err)
-	}
+	for name, c := range cases {
+		t.Run(name, func(t *testing.T) {
+			formattedTimeValue := FormatDateTime(c.TimeValue)
+			t.Logf("formated: %v -> %v", c.TimeValue, formattedTimeValue)
 
-	if e, a := refTime, parseTime; !e.Equal(a) {
-		t.Errorf("expected %v, got %v", e, a)
+			// Round Trip time value ensure format and parse are compatible.
+			parsedTimeValue, err := ParseDateTime(formattedTimeValue)
```

Comment:
Can ParseDateTime function do fomatting if required?

Reply:
clarified offline the Parse functions wouldn't need to perform formatting, since `time.Time` value returned is the "universal" format.